### PR TITLE
Connection infestation mechanics

### PR DESCRIPTION
**Pull request by cmavelis**
- directional progress for "roots" reaching out from fully infested dome
- tuning of infestation mechanics
  - root reaching a dome increases its random infestation chance
- BONUS: root recoil animated smoothly when progress set to 0

### CODE REVIEW

File: autoloads/dome_connections.gd [hunk 1]
```diff
@@ -5,67 +5,87 @@ var connections = []
...
+		
+		if c.infestation_progress >= 1 && c.dome_b.infestation_percentage == 0:
+			# TODO: this causes a Safe, "camped" dome to be considered instantly cleansed
+			if !c.dome_b.is_occupied():
+				c.dome_b.add_infestation(.0001)
+			# this gets spammed when a squad is sitting on a safe dome
+			print('CONNECTOR: spread infestation to ', c.dome_b.get_name())
+			# NOOOOTE: if fully infested while roots going out, will resey current progress
+			# TEST MECHANIC -- after seeding new infestation, "root" recedes almost all the way
+			c.infestation_progress = 0.001
```
Comment (cmavelis) [inline autoloads/dome_connections.gd:23]: need to revisit this tomorrow

Comment (cmavelis) [reply autoloads/dome_connections.gd]: sorted it out

**Comment by cmavelis** [comment]
everything in `testing/` can be removed if you prefer, it's just little experiments with directional progress bars

**Review by avelitj12 — APPROVED** [review]
Code LGTM. Merged into my branch to test integration and works well! Up to you if you want to remove the testing folder, I don't mind it.

**Pull request merged**